diff --git a/src/pattern.c b/src/pattern.c
index 0f30f68..2894ce0 100644
--- a/src/pattern.c
+++ b/src/pattern.c
@@ -54,6 +54,10 @@
 #include ATLAS_LAPACK_H
 #endif
 
+#ifdef HAVE_ACCELERATE
+#include ACCELERATE_H
+#endif
+
 #ifdef HAVE_LAPACKE
 #include LAPACKE_H
 #endif
@@ -125,7 +129,7 @@ patternstruct	*pattern_init(profitstruct *profit, pattypenum ptype, int ncomp)
     }
 
   return pattern;
-  }  
+  }
 
 
 /****** pattern_end ***********************************************************
@@ -219,6 +223,10 @@ void	pattern_fit(patternstruct *pattern, profitstruct *profit)
 /* Solve the system */
 #if defined(HAVE_LAPACKE)
   LAPACKE_dposv(LAPACK_COL_MAJOR, 'L', nvec, 1, alpha, nvec, beta, nvec);
+#elif defined(HAVE_ACCELERATE)
+  int info,n_beta=1;
+  char* upper_or_lower="U";
+  dposv_(upper_or_lower, &nvec, &n_beta, alpha, &nvec, beta, &nvec, &info);
 #else
   clapack_dposv(CblasRowMajor, CblasUpper, nvec, 1, alpha, nvec, beta, nvec);
 #endif
